src/opm/output/eclipse/AggregateUDQData.cpp

```diff
@@ -64,30 +64,423 @@ namespace {
         return inteHead[163];
     }
 
-     // Categorize function in terms of which token-types are used in formula
-    int define_type(const std::set<Opm::UDQTokenType> tokens) {
-        int type = -4;
+
+    // function to return true if token is a function
+    bool tokenTypeFunc(const Opm::UDQTokenType& token) {
+        bool type = false;
+        std::vector <Opm::UDQTokenType> type_1 = {
```

Comment:
Functions which closely resemble e.g. tokenTypeFunc is already implemented [here](https://github.com/OPM/opm-common/blob/master/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQEnums.hpp). Maybe some changes are not needed - but I am quite certain the functions already present in can be used for this.

Reply:
I have now modified code to use existing functions, however, still some additional functions are needed.